### PR TITLE
fix: make sure maintenance=true is honored properly

### PR DESCRIPTION
**Pull request by harshavardhana**
## Community Contribution License
All community contributions in this pull request are licensed to the project maintainers
under the terms of the [Apache 2 license](https://www.apache.org/licenses/LICENSE-2.0). 
By creating this pull request I represent that I have the right to license the 
contributions to the project maintainers under the Apache 2 license.

## Description
fix: make sure maintenance=true is honored properly

## Motivation and Context
fixes a regression from #18700

## How to test this PR?
```
minio server ~/test
curl http://localhost:9000/minio/health/cluster?maintenance=true (this must fail)
```

## Types of changes
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Optimization (provides speedup with no functional changes)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [x] Fixes a regression from #18700
- [ ] Unit tests added/updated
- [ ] Internal documentation updated
- [ ] Create a documentation update request [here](https://github.com/minio/docs/issues/new?label=doc-change,title=Doc+Updated+Needed+For+PR+github.com%2fminio%2fminio%2fpull%2fNNNNN)

### CODE REVIEW

File: cmd/erasure-server-pool.go [hunk 1]
```diff
@@ -2322,6 +2322,10 @@ func (z *erasureServerPools) Health(ctx context.Context, opts HealthOptions) Hea
 	storageInfo := z.StorageInfo(ctx, false)
 
 	for _, disk := range storageInfo.Disks {
+		if disk.Local && opts.Maintenance {
```
Comment (vadmeste) [inline cmd/erasure-server-pool.go:2325]: I am not sure why we are ignoring local disks here ? can`t a failed disk in the local node cause a wrong maintenance information ? Also not sure what bug will this fix

Comment (harshavardhana) [reply cmd/erasure-server-pool.go:2325]: Maintenance means I want to remove this node and do I have enough quorum to achieve that.

Read what we did earlier before the new code was implemented.

Comment (vadmeste) [reply cmd/erasure-server-pool.go:2325]: oh right, I understand now

**Review by vadmeste — APPROVED** [review]
LGTM

**Pull request merged**